llvm/lib/Target/AMDGPU/R600ISelLowering.cpp

```diff
@@ -100,6 +100,11 @@ R600TargetLowering::R600TargetLowering(const TargetMachine &TM,
 
   setOperationAction(ISD::FSUB, MVT::f32, Expand);
 
+  setOperationAction(ISD::IS_FPCLASS,
+                     {MVT::f32, MVT::v2f32, MVT::v3f32, MVT::v4f32, MVT::v5f32,
```

Comment:
Does it really support this many data types?

Reply:
Technically only need to set this for the minimum set of legal types, but it doesn't hurt to set it on illegal types that already defaulted to expand

Reply:
It turns out this is also working around some kind of bug in select / select_cc handling so I'm going to leave it alone